app/utility/oligrapher.rb

```diff
@@ -11,34 +11,47 @@ module Oligrapher
 
   def self.configuration(map:, current_user: nil)
     is_owner = current_user.present? && map.user_id == current_user.id
+    owner = map.user || current_user || nil
 
     {
       graph: map.graph_data.to_h,
-      settings: { debug: true },
-      display: { modes: { editor: is_owner } },
+      settings: { debug: Rails.env.development? },
       attributes: {
         id: map.id,
         title: map.title,
-        subtitle: '',
+        subtitle: map.description,
         date: (map.created_at || Time.current).strftime('%B %d, %Y'),
-        user: { name: current_user&.username },
-        settings: {
-          private: map.is_private,
-          clone: map.is_cloneable,
-          defaultStoryMode: false,
-          defaultExploreMode: true,
-          storyModeOnly: false,
-          exploreModeOnly: false
-        },
-        editors: map.usernames,
-        links: [
-          { text: 'Edit', url: 'https://littlesis.org/oligrapher/edit' },
-          { text: 'Clone', url: 'https://littlesis.org/oligrapher/clone' },
-          { text: 'Disclaimer', url: 'https://littlesis.org/oligrapher/disclaimer' }
-        ]
+        owner: owner.present? ? { id: owner.id, name: owner.username, url: owner.url } : nil,
+        user: current_user.present? ? { id: current_user.id, name: current_user.username } : nil,
+        settings: JSON.parse(map.settings || '{}').merge({ private: map.is_private, clone: map.is_cloneable }),
+        editors: is_owner ? editor_data(map) : confirmed_editor_data(map),
+        oligrapher_version: map.oligrapher_version
       } }
   end
 
+  def self.confirmed_editor_data(map)
+    User
+      .where(id: map.confirmed_editor_ids)
+      .map { |u|
+        {
+          name: u.username,
+          url: u.url
+        }
+      }
+  end
+
+  def self.editor_data(map)
+    map.editors.map { |e|
```

Comment:
Although I don't expect a map to have to that many editors, this executes an individual SQL statements for each `User`. It's better to do the same strategy as `confirmed_editor_data` where you get all users in a single SQL query.